bokehjs/src/coffee/models/tools/util.ts

```diff
@@ -0,0 +1,33 @@
+import {Renderer} from "../renderers/renderer"
+import {GlyphRenderer} from "../renderers/glyph_renderer"
+import {GraphRenderer} from "../renderers/graph_renderer"
+
+import {includes} from "core/util/array"
+
+export type DataRenderer = GlyphRenderer | GraphRenderer
+
+export type RendererSpec = DataRenderer[] | "auto" | null
+
+export function compute_renderers(renderers: RendererSpec, all_renderers: Renderer[], names: string[]): DataRenderer[] {
+    if (renderers == null) {
+      return []
+    }
+
+    if (renderers.length == 0) {
+      return []
+    }
```

Comment:
Can't we just have one value for "empty" behavior (drop `null`)?

Reply:
Do we have a technical way to validate against `None` values? If not I'd rather have it do something defined and documented (which is the only reason I added a path for it)